### PR TITLE
Async `browser.newPage`

### PR DESCRIPTION
**Pull request by inancgumus**
## What?

#1294

## Checklist

- [x] I have performed a self-review of my code
- [x] I have added tests for my changes
- [x] I have commented on my code, particularly in hard-to-understand areas

### CODE REVIEW

File: tests/browser_context_test.go [hunk 1]
```diff
@@ -657,28 +657,31 @@ func TestK6Object(t *testing.T) {
 
 			// First test with browser.newPage
 			got, err := vu.TestRT.RunOnEventLoop(`
-				const p = browser.newPage();
-				p.goto("about:blank");
-				const o = p.evaluate(() => window.k6);
-				JSON.stringify(o);
+				(async function() {
+					const p = await browser.newPage();
+					p.goto("about:blank");
```
Comment (ankur22) [inline tests/browser_context_test.go:662]: I know it's not from your change, but it looks like the `goto` needs an `await`

File: tests/browser_context_test.go [hunk 2]
```diff
@@ -657,28 +657,31 @@ func TestK6Object(t *testing.T) {
...
 
 			// Now test with browser.newContext
 			got, err = vu.TestRT.RunOnEventLoop(`
-				const test = async function() {
+				(async function() {
 					await browser.closeContext();
 					const c = await browser.newContext();
-					const p2 = c.newPage();
+					const p2 = await c.newPage();
 					p2.goto("about:blank");
```
Comment (ankur22) [inline tests/browser_context_test.go:678]: Same here, missing `await` on `goto`.

File: tests/browser_context_test.go [hunk 3]
```diff
@@ -704,17 +707,18 @@ func TestNewTab(t *testing.T) {
...
-		const p = browser.newPage()
-		p.goto("%s/%s/ping.html")
-		
-		const p2 = browser.context().newPage()
-		p2.goto("%s/%s/ping.html")
-	`, s.URL, testBrowserStaticDir, s.URL, testBrowserStaticDir))
+	_, err := vu.TestRT.RunOnEventLoop(fmt.Sprintf(`
+		(async function() {
+			const p = await browser.newPage()
+			p.goto("%s/%s/ping.html")
```
Comment (ankur22) [inline tests/browser_context_test.go:717]: And in here 😅

**Review by ankur22 — APPROVED** [review]
LGTM

File: tests/browser_context_test.go (hunk 3 above)
Comment (inancgumus) [reply tests/browser_context_test.go:717]: Ah, nice catch 👏 I'll send another PR for this to not to mess with the chain of other async PRs 🥲

**Pull request merged**